src/Interpreters/Context.h

```diff
@@ -576,6 +576,9 @@ class Context: public ContextData, public std::enable_shared_from_this<Context>
     std::optional<UUID> getUserID() const;
     String getUserName() const;
 
+    /// WARNING: all requests with this context won't trigger any access checks.
+    void resetUser();
```

Comment:
Why do you need this? Just use the global context. It's nice that we have only one global context.

Reply:
I don't want to completely override the original local context. What if it had some other stuff in it?

Reply:
I think it's better to completely override the original context always when it's not SQL_SECURITY=INVOKER.
Because we want a context created for the definer, not kind of a mixture between contexts for two users. I mean the all the settings should the definer's settings, and roles should be the definer's roles.

Reply:
Probably we need to keep the information about the initial query in the `ClientInfo` of the new context though.